regression-test/suites/vault_p0/default/test_default_vault.groovy

```diff
@@ -0,0 +1,134 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_default_vault", "nonConcurrent") {
+    if (!isCloudMode()) {
+        logger.info("skip ${name} case, because not cloud mode")
+        return
+    }
+
+    if (!enableStoragevault()) {
+        logger.info("skip ${name} case")
+        return
+    }
+
+    try {
+        sql """ UNSET DEFAULT STORAGE VAULT; """
+
+        expectExceptionLike({
+            sql """ set not_exist as default storage vault """
+        }, "invalid storage vault name")
+
+        def tableName = "table_use_vault"
+        sql "DROP TABLE IF EXISTS ${tableName}"
+
+        expectExceptionLike({
+            sql """
+                CREATE TABLE ${tableName} (
+                    `key` INT,
+                    value INT
+                ) DUPLICATE KEY (`key`) DISTRIBUTED BY HASH (`key`) BUCKETS 1
+                PROPERTIES ('replication_num' = '1')
+            """
+        }, "No default storage vault")
+
+        sql """
+            CREATE STORAGE VAULT IF NOT EXISTS create_s3_vault_for_default
+            PROPERTIES (
+                "type"="S3",
+                "s3.endpoint"="${getS3Endpoint()}",
+                "s3.region" = "${getS3Region()}",
+                "s3.access_key" = "${getS3AK()}",
+                "s3.secret_key" = "${getS3SK()}",
+                "s3.root.path" = "create_s3_vault_for_default",
+                "s3.bucket" = "${getS3BucketName()}",
+                "s3.external_endpoint" = "",
+                "provider" = "${getS3Provider()}",
+                "set_as_default" = "true"
+            );
+        """
+
+        sql """ set create_s3_vault_for_default as default storage vault """
+        def vaultInfos = sql """ SHOW STORAGE VAULT """
+        // check if create_s3_vault_for_default is set as default
+        for (int i = 0; i < vaultInfos.size(); i++) {
+            def name = vaultInfos[i][0]
+            if (name.equals("create_s3_vault_for_default")) {
+                // isDefault is true
+                assertEquals(vaultInfos[i][3], "true")
+            }
+        }
+
+        sql """ UNSET DEFAULT STORAGE VAULT; """
+        vaultInfos = sql """ SHOW STORAGE VAULT """
+        for (int i = 0; i < vaultInfos.size(); i++) {
+            assertEquals(vaultInfos[i][3], "false")
+        }
+
+
+        sql """ set built_in_storage_vault as default storage vault """
+
+        sql "DROP TABLE IF EXISTS ${tableName} FORCE;"
+        sql """
+            CREATE TABLE ${tableName} (
+                `key` INT,
+                value INT
+            ) DUPLICATE KEY (`key`) DISTRIBUTED BY HASH (`key`) BUCKETS 1
+            PROPERTIES ('replication_num' = '1')
+        """
+
+        sql """ insert into ${tableName} values(1, 1); """
+        qt_sql """ select * from ${tableName}; """
```

Comment:
just check the result of the select stmt in memory 
we dont need to  manage another file called test_default_vault.out
```
result =  sql """ select * ..."""
assertEqual(result.size(), 1);
assertEqual(result[0][0], 1);
...

Reply:
> just check the result of the select stmt in memory we dont need to manage another file called test_default_vault.out
> 
> ```
> result =  sql """ select * ..."""
> assertEqual(result.size(), 1);
> assertEqual(result[0][0], 1);
> ...
> ```

done